modules/navier_stokes/src/functormaterials/INSFVEnthalpyFunctorMaterial.C

```diff
@@ -32,31 +33,106 @@ INSFVEnthalpyFunctorMaterial::validParams()
+
+    addFunctorProperty<ADReal>(NS::time_deriv(getParam<MooseFunctorName>(NS::specific_enthalpy)),
+                               [this](const auto & r, const auto & t)
+                               { return _cp(r, t) * _temperature.dot(r, t); });
+
+    addFunctorProperty<ADReal>(
+        "rho_cp_temp", [&rho_h](const auto & r, const auto & t) -> ADReal { return rho_h(r, t); });
+
+    addFunctorProperty<ADReal>("cp_temp",
+                               [&h](const auto & r, const auto & t) -> ADReal { return h(r, t); });
+  }
+  else if (_h)
+  {
+    const auto & rho_h = addFunctorProperty<ADReal>(NS::enthalpy_density,
+                                                    [this](const auto & r, const auto & t)
+                                                    { return _rho(r, t) * (*_h)(r, t); });
+
+    addFunctorProperty<ADReal>(NS::time_deriv(getParam<MooseFunctorName>(NS::specific_enthalpy)),
+                               [this](const auto & r, const auto & t) { return _h->dot(r, t); });
+
+    addFunctorProperty<ADReal>(
+        "rho_cp_temp", [&rho_h](const auto & r, const auto & t) -> ADReal { return rho_h(r, t); });
+
+    addFunctorProperty<ADReal>(
+        "cp_temp", [this](const auto & r, const auto & t) -> ADReal { return (*_h)(r, t); });
+  }
+  else
+  {
+    const auto & rho_h = addFunctorProperty<ADReal>(
+        NS::enthalpy_density,
+        [this](const auto & r, const auto & t)
+        { return _rho(r, t) * _fp->h_from_p_T((*_pressure)(r, t), _temperature(r, t)); });
 
-  const auto & h = addFunctorProperty<ADReal>(NS::specific_enthalpy,
-                                              [this](const auto & r, const auto & t)
-                                              { return _cp(r, t) * _temperature(r, t); });
+    const auto & h =
+        addFunctorProperty<ADReal>(NS::specific_enthalpy,
+                                   [this](const auto & r, const auto & t) {
+                                     return _fp->h_from_p_T((*_pressure)(r, t), _temperature(r, t));
+                                   });
 
-  addFunctorProperty<ADReal>(NS::time_deriv(getParam<MooseFunctorName>(NS::specific_enthalpy)),
-                             [this](const auto & r, const auto & t)
-                             { return _cp(r, t) * _temperature.dot(r, t); });
+    addFunctorProperty<ADReal>(
+        NS::time_deriv(getParam<MooseFunctorName>(NS::specific_enthalpy)),
+        [this](const auto & r, const auto & t)
+        {
+          Real h, dh_dp, dh_dT;
+          _fp->h_from_p_T((*_pressure)(r, t).value(), _temperature(r, t).value(), h, dh_dp, dh_dT);
+          return dh_dT * _temperature.dot(r, t) + dh_dp * _pressure->dot(r, t);
+        });
 
-  addFunctorProperty<ADReal>(
-      "rho_cp_temp", [&rho_h](const auto & r, const auto & t) -> ADReal { return rho_h(r, t); });
+    addFunctorProperty<ADReal>(
+        "rho_cp_temp", [&rho_h](const auto & r, const auto & t) -> ADReal { return rho_h(r, t); });
```

Comment:
If we are making this more general, I'd not call it `rho_cp_temp` anymore. Most of the code fetches this under the hood so I hope not a lot of input files need changing.

Reply:
ok let's just not define these functors anymore. 

added the transient test too